android/sdl_android/src/main/java/com/smartdevicelink/managers/video/VideoStreamManager.java

```diff
@@ -284,45 +363,95 @@ public void onError(String info) {
-        if (majorProtocolVersion >= 5) {
-            if (internalInterface.getSystemCapabilityManager() != null) {
-                final VideoStreamingParameters params = ( parameters == null) ? new VideoStreamingParameters() : new VideoStreamingParameters(parameters);
-                internalInterface.getSystemCapabilityManager().getCapability(SystemCapabilityType.VIDEO_STREAMING, new OnSystemCapabilityListener() {
-                    @Override
-                    public void onCapabilityRetrieved(Object capability) {
-                        params.update((VideoStreamingCapability) capability, vehicleMake);    //Streaming parameters are ready time to stream
-                        startStreaming(params, encrypted);
-                    }
+        if (!HMILevel.HMI_NONE.equals(currentOnHMIStatus.getHmiLevel()) && VideoStreamManager.this.parameters == null) {
+            getVideoStreamingParams();
+        }
+        checkState();
+        processCapabilitiesWithPendingStart(encrypted, parameters);
+    }
 
-                    @Override
-                    public void onError(String info) {
-                        stateMachine.transitionToState(StreamingStateMachine.ERROR);
-                        DebugTool.logError(TAG, "Error retrieving video streaming capability: " + info);
-                    }
-                }, false);
+    private void configureGlobalParameters(Context context, Class<? extends SdlRemoteDisplay> remoteDisplayClass, boolean encrypted, List<VideoStreamingRange> listOfStreamingRange) {
+        this.context = new WeakReference<>(context);
+        this.remoteDisplayClass = remoteDisplayClass;
+        this.isEncrypted = encrypted;
+        this.majorProtocolVersion = internalInterface.getProtocolVersion().getMajor();
+        if (listOfStreamingRange != null) {
+            this.listOfStreamingRanges = listOfStreamingRange;
+        }
+    }
+//
+//    // regardless of VideoStreamingParameters are specified or not, we should refer to VideoStreamingCapability.
+//        if (majorProtocolVersion >= 5) {
+//        if (internalInterface.getSystemCapabilityManager() != null) {
+//            final VideoStreamingParameters params = ( parameters == null) ? new VideoStreamingParameters() : new VideoStreamingParameters(parameters);
+//            internalInterface.getSystemCapabilityManager().getCapability(SystemCapabilityType.VIDEO_STREAMING, new OnSystemCapabilityListener() {
+//                @Override
+//                public void onCapabilityRetrieved(Object capability) {
+//                    params.update((VideoStreamingCapability) capability, vehicleMake);    //Streaming parameters are ready time to stream
+//                    startStreaming(params, encrypted);
+//                }
+
+    private void processCapabilitiesWithPendingStart(boolean encrypted, VideoStreamingParameters parameters){
+        if (parameters == null) {
```

Comment:
Currently with this implementation we do not update the existing parameters with the capabilities received. This was the goal of [SDL 0323](https://github.com/smartdevicelink/sdl_java_suite/issues/1569)
If the parameters sent to this method are not null we still want to update them based on the capabilities received.
I propose removing the check if parameters are null `if (parameters == null) {` and create `params` just before calling `getCapability()`

`final VideoStreamingParameters params = ( parameters == null) ? new VideoStreamingParameters() : new VideoStreamingParameters(parameters);`

Reply:
Hi @RHenigan, in terms of SDL 0323, your suggested change above looks good to me. Thanks.

Reply:
@RHenigan could you please check if the suggested implementation for this case is suitable?